ironfish/src/primitives/unsignedTransaction.ts

```diff
@@ -55,4 +61,40 @@ export class UnsignedTransaction {
 
     return result
   }
+
+  descriptions(
+    incomingViewKey: string,
+    outgoingViewKey: string,
+  ): {
+    receivedNotes: Note[]
+    spentNotes: Note[]
+    mints: NativeMintDescription[]
+    burns: NativeBurnDescription[]
+  } {
+    const descriptions = this.takeReference().descriptions()
+    this.returnReference()
+
+    const receivedNotes = []
+    const spentNotes = []
+    for (const serializedOutput of descriptions.outputs) {
+      const note = new NoteEncrypted(serializedOutput)
+
+      const receivedNote = note.decryptNoteForOwner(incomingViewKey)
+      if (receivedNote) {
+        receivedNotes.push(receivedNote)
+      }
+
+      const spentNote = note.decryptNoteForSpender(outgoingViewKey)
+      if (spentNote) {
+        spentNotes.push(spentNote)
+      }
```

Comment:
Wondering if we'll want/need to handle notes showing up in both received and spent when we render

Reply:
Yeah. I thought I'd just return everything here for now and then the caller can decide what to do.